src/features/events/hooks/useEventMutations.ts

```diff
@@ -0,0 +1,121 @@
+import { ZetkinEvent } from 'utils/types/zetkin';
+import {
+  eventCreate,
+  eventCreated,
+  eventDeleted,
+  eventUpdate,
+  eventUpdated,
+} from '../store';
+import { useApiClient, useAppDispatch } from 'core/hooks';
+
+export type ZetkinEventPatchBody = Partial<
+  Omit<
+    ZetkinEvent,
+    'id' | 'activity' | 'campaign' | 'location' | 'organization'
+  >
+> & {
+  activity_id?: number | null;
+  campaign_id?: number;
+  cancelled?: string | null;
+  contact_id?: number | null;
+  location_id?: number | null;
+  organization_id?: number;
+  published?: string | null;
+};
+
+export type ZetkinEventPostBody = ZetkinEventPatchBody;
+
+type useEventMutationsReturn = {
+  cancelEvent: (eventId: number) => void;
+  createEvent: (eventBody: ZetkinEventPostBody) => Promise<ZetkinEvent>;
+  deleteEvent: (eventId: number) => void;
+  publishEvent: (eventId: number) => void;
+  restoreEvent: (eventId: number) => void;
+  setPublished: (eventId: number, published: string | null) => void;
+  setTitle: (eventId: number, title: string) => void;
+  setType: (eventId: number, id: number | null) => void;
+  updateEvent: (eventId: number, data: ZetkinEventPatchBody) => void;
```

Comment:
There's a red flag here IMO, which is that all of these mutations take the `eventId` except for one. That makes me think that `createEvent()` doesn't belong here, and that all other mutations should be getting their ID from their closure (the ID should be passed to `useEventMutations()` instead).